topdown/graphql.go

```diff
@@ -170,46 +237,60 @@ func pruneIrrelevantGraphQLASTNodes(value ast.Value) ast.Value {
 
 // Reports errors from parsing/validation.
 func builtinGraphQLParse(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
-	// Get the raw strings from each operand so that we can
-	// feed them to the GraphQL parser functions.
-	rawQuery, err := builtins.StringOperand(operands[0].Value, 1)
-	if err != nil {
-		return err
-	}
-	rawSchema, err := builtins.StringOperand(operands[1].Value, 1)
-	if err != nil {
-		return err
+	var queryDoc *gqlast.QueryDocument
+	var schemaDoc *gqlast.SchemaDocument
+	var err error
+
+	// Parse/translate query if it's a string/object.
+	switch x := operands[0].Value.(type) {
+	case ast.String:
+		queryDoc, err = parseQuery(string(x))
+		if err != nil {
+			return err
+		}
+	case ast.Object:
+		queryDoc, err = objectToQueryDocument(x)
+		if err != nil {
+			return err
+		}
+	default:
+		// Error if wrong type.
+		return builtins.NewOperandTypeErr(0, x, "string", "object")
 	}
```

Comment:
```suggestion
	// Parse/translate query if it's a string/object.
	switch x := operands[0].Value.(type) {
	case ast.String:
		queryDoc, err = parseQuery(string(x))
	case ast.Object:
		queryDoc, err = objectToQueryDocument(x)
	default:
		// Error if wrong type.
		return builtins.NewOperandTypeErr(0, x, "string", "object")
	}
	if err != nil {
		return err
	}
```
debatable... it doesn't matter I think.

Reply:
I think it's a little cleaner looking, so I rolled the change into my latest updates. :+1: